xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/doc/XWikiDocumentTest.java

```diff
@@ -34,8 +34,12 @@
 import java.util.Set;
 import java.util.Vector;
 
+import javax.servlet.http.HttpServletRequest;
+
 import org.apache.velocity.VelocityContext;
+import org.jmock.Expectations;
 import org.jmock.Mock;
+import org.jmock.Mockery;
```

Comment:
How much is covered of the new code? There doesn't seem to be many tests compared to all the code added? I'm worried about that since we're touching the core here, we need to be careful not to introduce bugs.